src/libstore/build/derivation-goal.cc

```diff
@@ -988,13 +988,28 @@ void DerivationGoal::resolvedFinished()
             auto resolvedHash = get(resolvedHashes, wantedOutput);
             if ((!initialOutput) || (!resolvedHash))
                 throw Error(
-                    "derivation '%s' doesn't have expected output '%s' (derivation-goal.cc/resolvedFinished,resolve)",
+                    "derivation '%s' doesn't have expected output '%s'",
                     worker.store.printStorePath(drvPath), wantedOutput);
             auto realisation = get(resolvedResult.builtOutputs, DrvOutput { *resolvedHash, wantedOutput });
```

Comment:
example failure (reported by @trofi):
```
error: derivation '/nix/store/g7q24qrdzdcvsirxw8aw0alc9s0qsl77-shared-mime-info-unstable-2021-12-03.drv' doesn't have expected output 'dev @ sha256:094p4p14vbmqhhd2g4nr08phln6hr789rzqnjwll9q79kisc7n7g'
        builtOutputs:
                out @ sha256:094p4p14vbmqhhd2g4nr08phln6hr789rzqnjwll9q79kisc7n7g (key) = /nix/store/98f30ch5mldzkw0zznx2s2ll6r940jv3-shared-mime-info-unstable-2021-12-03
```
=> hash matches, but the wanted output is not part of the built outputs.